rustls/src/common_state.rs

```diff
@@ -493,6 +566,50 @@ impl CommonState {
         self.send_warning_alert_no_log(AlertDescription::CloseNotify);
     }
 
+    pub(crate) fn eager_send_close_notify(
```

Comment:
This fn looks like it shares a decent amount of code with `eager_send_some_plaintext` - is there a way to reduce some of the duplication?